core/dbt/events/functions.py

```diff
@@ -281,7 +286,15 @@ def send_exc_to_logger(
 # (i.e. - mutating the event history, printing to stdout, logging
 # to files, etc.)
 def fire_event(e: Event) -> None:
-    # TODO manage history in phase 2:  EVENT_HISTORY.append(e)
+
+    # if and only if the event history deque will be completely filled by this event
+    # fire warning that old events are now being dropped
+    global EVENT_HISTORY
+    if len(EVENT_HISTORY) == (100000 - 1):
+        EVENT_HISTORY.append(e)
+        fire_event(EventBufferFull())
```

Comment:
what if someone is running on a machine so small that memory runs out before this hard-coded limit? would we never log it?

Reply:
I'm ok looking into this after the RC3 since I feel like it will be a rare case